src/mscorlib/src/System/ThrowHelper.cs

```diff
@@ -122,7 +122,7 @@ internal static void ThrowAddingDuplicateWithKeyArgumentException(object key)
 
         internal static void ThrowKeyNotFoundException(object key)
         {
-            throw new KeyNotFoundException(key.ToString());
+            throw new KeyNotFoundException(SR.Format(SR.Arg_KeyNotFoundWithKey, key.ToString()));
```

Comment:
Could you please change this to:

```
internal static void ThrowKeyNotFoundException(object key)
{
    throw GetKeyNotFoundException(object key);
}
```

to be consistent with the other similar methods? It will let the JIT to generate better code.

Reply:
@jkotas I understand why the ThrowXX method is valuable - the comment explains. Why is it found helpful to further break out constructing the exception?

Reply:
JIT will look into small methods to observe what they do. If the method always throws exception, the JIT will move the call to it to the "cold" section of the code and it will also know that the method will never return.

Breaking out constructing of the exception allows this optimization to kick in. The exception construction is large enough piece of code to prevent the optimization from kicking in.

Check the discussion in #7580 that introduced this.